### PR TITLE
Add space access report

### PR DESCRIPTION
**Pull request by LowBudgetMan**
## Issue
Connects #368 

## What was done
- [x] Add endpoint for getting all spaces and users who belong to those spaces
- [x] Add endpoint for getting all users

## How to test
1. Using Postman, connect to /api/reports/space
2. Using Postman, connect to /api/reports/user

### CODE REVIEW

File: api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt [hunk 1]
```diff
@@ -167,31 +171,61 @@ class ReportGeneratorControllerTest {
...
 
         val actualSpaceReport = objectMapper.readValue<List<SpaceReportItem>>(
-            result.response.contentAsString,
-            objectMapper
-                .typeFactory
-                .constructCollectionType(MutableList::class.java, SpaceReportItem::class.java)
+                result.response.contentAsString,
+                objectMapper
+                        .typeFactory
+                        .constructCollectionType(MutableList::class.java, SpaceReportItem::class.java)
```
Comment (RandomAshes) [inline api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt:192]: Inconsistent spacing?

**Review by michaelrygiel — APPROVED** [review]
❗ 🗑️ 🔥

**Review by RandomAshes — DISMISSED** [review]
LGTM. Pulled down and endpoints work as expected.

File: api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt (hunk 1 above)
Comment (egriff38ford) [reply api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt:192]: Those are different objects. We could probably pull that out though

File: api/src/main/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorController.kt [hunk 2]
```diff
@@ -17,18 +17,57 @@
...
+
+        return ResponseEntity.status(HttpStatus.FORBIDDEN).build()
     }
 
+    @GetMapping("/user")
+    fun getUserReport():ResponseEntity<List<String>> {
+        val userName: String = SecurityContextHolder.getContext().authentication.name.toLowerCase()
+
+        val authorizedUsers = users.toLowerCase().split(",")
+        val isAuthorizedUser = authorizedUsers.contains(userName)
```
Comment (egriff38ford) [inline api/src/main/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorController.kt:65]: Duplication ^^

**Review by willot — APPROVED** [review]

**Review by RandomAshes — APPROVED** [review]
LGTM!

File: api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt (hunk 1 above)
Comment (RandomAshes) [reply api/src/test/kotlin/com/ford/internalprojects/peoplemover/report/ReportGeneratorControllerTest.kt]: I was just noting that on line 189 and 190 is like 2 spaces vs 191 and 192 is 4 it looks like..

**Pull request merged**